File: spring-boot-angular-js/spring-boot-angular-js/src/main/resources/public/js/companyService.js
app.service('companyService', function($http) {
	delete $http.defaults.headers.common['X-Requested-With'];

	this.getAllCompanies = function() {
		return $http({
			method : 'GET',
			url : '/companies'
		});
	};

	this.addCompany = function(company) {
		return $http({
			method : 'PUT',
			url : '/companies',
			headers : {
				'Content-Type' : 'application/json'
			},
			data : company
		});
	};

	this.updateCompany = function(companyID) {
		return $http({
			method : 'POST',
			url : '/companies/' + companyID
		});
	};

	this.getCompany = function(companyID) {
		return $http({
			method : 'GET',
			url : '/companies/' + companyID
		});
	};
});

File: spring-boot-angular-js/spring-boot-angular-js/src/main/java/com/spring/boot/angular/service/CompanyService.java
package com.spring.boot.angular.service;

import java.util.List;

import com.spring.boot.angular.model.Company;

public interface CompanyService {

	List<Company> getAllCompanies();

	Company createCompany(Company company);

	Company updateCompany(Integer companyId, Company company);

	Company getCompany(Integer companyId);

}
